src/soundio/soundmanagerutil.h

```diff
@@ -56,46 +56,46 @@ class AudioPath {
     /// methods including getStringFromType, isIndexed, getTypeFromInt,
     /// channelsNeededForType (if necessary), the subclasses' getSupportedTypes
     /// (if necessary), etc.
-    enum AudioPathType {
-        MASTER,
-        HEADPHONES,
-        BOOTH,
-        BUS,
-        DECK,
-        VINYLCONTROL,
-        MICROPHONE,
-        AUXILIARY,
-        RECORD_BROADCAST,
-        INVALID, // if this isn't last bad things will happen -bkgood
-    };
-    AudioPath(unsigned char channelBase, unsigned char channels);
-    virtual ~AudioPath() = default;
-    AudioPathType getType() const;
-    ChannelGroup getChannelGroup() const;
-    unsigned char getIndex() const;
-    bool channelsClash(const AudioPath &other) const;
-    QString getString() const;
-    static QString getStringFromType(AudioPathType type);
-    static QString getTrStringFromType(AudioPathType type, unsigned char index);
-    static AudioPathType getTypeFromString(QString string);
-    static bool isIndexed(AudioPathType type);
-    static AudioPathType getTypeFromInt(int typeInt);
-
-    /// Returns the minimum number of channels needed on a sound device for an
-    /// AudioPathType.
-    static unsigned char minChannelsForType(AudioPathType type);
-
-    // Returns the maximum number of channels needed on a sound device for an
-    // AudioPathType.
-    static unsigned char maxChannelsForType(AudioPathType type);
-
-    uint hashValue() const {
+  enum class AudioPathType : int {
```

Comment:
any particular reason for specifying the underlying type explicitly?

Reply:
> any particular reason for specifying the underlying type explicitly?

There is code that uses the enum's `int` value and converts between the types. Although the underlying type of a scoped `enum class` always `int` if not otherwise specified, I though that explicit is better than implicit and shows that we actually want to use `int` here.

Reply:
I think its only important to specify the underlying type when doing a bit_cast/reinterpret_cast on a struct that contains this. Otherwise the type is only specified to be as many bits wide as required to hold all enum values... 

Just because we static_cast from and to `int` for this enum doesn't really mean that it needs to be represented in memory as an `int`... Anyways, I don't really see the usage warranted here, but idc. Just my two cents.